src/pepr/operator/controllers/network/defaults/allow-egress-dns.ts

```diff
@@ -0,0 +1,16 @@
+import { Direction } from "../../../crd";
+import { generate } from "../generate";
+
+export const allowEgressDNS = (namespace: string) => {
+  const netPol = generate(namespace, {
```

Comment:
yeah i originally had it as an opt-out, but dropped that since BB wasn't doing it to simplify the CR....will think about that one though, it's a valid point.

Reply:
Might be worth polling folks clusters to see if `k8s-app: kube-dns` is consistent across distros as a pod label. From a quick scan.... k3d, rke2, eks, and kind all use that label and place coredns in `kube-system`. Based on k8s docs pretty sure kubeadm does the same.

Potentially could use that as a default and if we want to have a safety net validate that selector returns a pod with a get or just expose a way to override? I don't recall offhand if there was a specific cluster that BB had in mind that used non-standard labels/ns or if it was always to err on the side of caution.

Reply:
If chatgpt doesnt lie-- kube-dns selector looks safe across most distros. Here's it working with K3d. Notice the default lookup works, but trying to use an alternate dns server fails.

![Screenshot 2024-01-12 at 7 19 36 PM](https://github.com/defenseunicorns/uds-core/assets/882485/db7db782-e477-4800-a9d3-410ef3d7fb99)

Reply:
same pod when you delete the CR (netpols are removed)

![Screenshot 2024-01-12 at 7 22 30 PM](https://github.com/defenseunicorns/uds-core/assets/882485/29d5e679-d0c8-4661-bc9b-0f2fcd15f927)